tests/linpde_gp/linfunctls/test_flattened.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+import numpy as np
+
+import linpde_gp
+from linpde_gp.linfunctls import DiracFunctional, FlattenedLinearFunctional
+
+def test_flat_stays_unchanged():
+    X = np.array([1., 2., 3.])
+    f = linpde_gp.functions.Zero((), ())
+
+    D = DiracFunctional((), (), X)
+    D_res = D(f)
+
+    L = FlattenedLinearFunctional(D)
+    L_res = L(f)
+
+    assert L_res.shape == D_res.shape
+    np.testing.assert_allclose(L_res, D_res)
+
+def test_tensor_becomes_flat():
+    I, O = 2, 3
+    N = 5
+    X = np.zeros((N, I))
+    f = linpde_gp.functions.Zero((I,), (O,))
```

Comment:
Can we use a nonzero `Constant` function here? Something like a reshaped `arange` would reveal more errors